compiler/rustc_trait_selection/src/traits/misc.rs

```diff
@@ -131,3 +133,19 @@ pub fn type_allowed_to_implement_copy<'tcx>(
 
     Ok(())
 }
+
+pub fn check_const_param_definitely_unequal<'tcx>(
```

Comment:
I think the name makes it confusing whether the `Ok` encodes "definitely unequal" or "could be equal". Can you change the name to something like:

```suggestion
pub fn check_tys_may_equate<'tcx>(
```

(maybe you can stick `for_const_param` or something if you want to discourage others from using this in other places... though it would be nice to use this in a `can_eq` type situation now that I think of it, since post-lazy-norm, `can_eq` for a projection vs rigid is basically useless, since it'll return "yes" with a nested normalizes-to obligation..)